pkg/state/state.go

```diff
@@ -1534,16 +1642,40 @@ func (s *stateManager) addBlocks() (*proto.Block, error) {
 		if err := s.stor.hitSources.appendBlockHitSource(block, blockchainCurHeight+1, hs); err != nil {
 			return nil, err
 		}
+		if !patch.isInitialized() {
+			if lErr := patch.load(block.BlockID(), s.stor.patches); lErr != nil {
+				return nil, wrapErr(RetrievalError, lErr)
+			}
+		}
+		fixSnapshotsToInitialHash := patch.take()
 		// Save block to storage, check its transactions, create and save balance diffs for its transactions.
-		if addErr := s.addNewBlock(block, lastAppliedBlock, chans, blockchainCurHeight, sh); addErr != nil {
+		addErr := s.addNewBlock(block, lastAppliedBlock, chans, blockchainCurHeight, fixSnapshotsToInitialHash, sh)
+		if addErr != nil {
 			return nil, addErr
 		}
+		blockchainCurHeight++ // we've just added a new block, so we have a new height
 
-		if s.needToFinishVotingPeriod(blockchainCurHeight + 1) {
+		if s.needToFinishVotingPeriod(blockchainCurHeight) {
 			// If we need to finish voting period on the next block (h+1) then
```

Comment:
I don't get all those hight manipulations. The `blockchainCurHeight + 1` is the hight of current block. So we checking if the voting period ended on previous block (on blockchain height) we set the protobuf activation flag on for the current block and following blocks. Correct? 
But how it was deserialised without errors on line 1606 if this block is a protobuf block and the flag was not set?

Reply:
And the comment is not correct then.